src/openrct2/ride/TrackDesign.cpp

```diff
@@ -843,16 +843,12 @@ void track_design_mirror(TrackDesign* td6)
     track_design_mirror_scenery(td6);
 }
 
-static void track_design_add_selection_tile(int16_t x, int16_t y)
+static void track_design_add_selection_tile(const CoordsXY& coords)
 {
-    for (const auto& tile : gMapSelectionTiles)
-    {
-        if (tile.x == x && tile.y == y)
-        {
-            return;
-        }
-    }
-    gMapSelectionTiles.push_back(CoordsXY(x, y));
+    auto tileIterator = std::find(gMapSelectionTiles.begin(), gMapSelectionTiles.end(), coords);
+    if (tileIterator == gMapSelectionTiles.end())
+        return;
+    gMapSelectionTiles.push_back(coords);
```

Comment:
```suggestion
    if (tileIterator == gMapSelectionTiles.end())
    {
        gMapSelectionTiles.push_back(coords);
    }
```

It should add when it doesn't find any already